gateway/src/config_parser.rs

```diff
@@ -113,6 +113,17 @@ impl<'c> Config<'c> {
             .into_iter()
             .map(|(name, config)| config.load(&base_path, name.clone()).map(|c| (name, c)))
             .collect::<Result<HashMap<String, StaticToolConfig>, Error>>()?;
+        // Validate that all tools have unique names.
+        let mut seen_names = HashSet::new();
+        for tool_config in tools.values() {
+            let tool_name = &tool_config.name;
+            if !seen_names.insert(tool_name) {
+                return Err(ErrorDetails::Config {
+                    message: format!("Duplicate tool name: '{}' in tool configs", tool_name),
+                }
+                .into());
+            }
+        }
```

Comment:
The _name_ can be duplicated. In fact that's why we want this change - today the _ID_ (`[tools.xxx]`) can't be duplicated, but sometimes you might want to define multiple tools with the same _name_ (for different scenarios, e.g. with different schemas).

Could you remove this check and update the tests?